subsys/partition_manager/CMakeLists.txt

```diff
@@ -50,17 +50,22 @@ endif()
 get_property(img_pm_subsys GLOBAL PROPERTY PM_SUBSYS)
 if (img_pm_subsys)
   foreach (subsys_pm_yml ${img_pm_subsys})
-    file(RELATIVE_PATH rel_path_to_yml ${ZEPHYR_BASE} ${subsys_pm_yml})
-    set(subsys_pm_preprocessed
-      ${PROJECT_BINARY_DIR}/${rel_path_to_yml})
+
+    # Each entry is "<absolute path to pm.yml> <build output path>", so in
+    # order to split on the space we listify the item, and use list(GET)
+    string(REPLACE " " ";" listified ${subsys_pm_yml})
+    list(GET listified 0 pm_yml_path)
+    list(GET listified 1 output_pm_yml_path)
```

Comment:
This comment follows the one here: https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/292#discussion_r409371392

If they always follow double, then we could consider making two lists, and then loop them similar to this:
https://github.com/tejlmand/zephyr/blob/zephyr_sdk_improvement/cmake/toolchain/zephyr/host-tools.cmake#L61-L69

then we would avoid regex here, as well as be able to witch to `foreach(pm_yml_path output_pm_yml_path IN ZIP_LISTS img_pm_subsys <second_list>)` as soon as minimum required CMake becomes 3.17.

Reply:
following the comment here:
```
get_property(img_pm_subsys_path GLOBAL PROPERTY PM_SUBSYS_PATH)
get_property(img_pm_subsys_output_path GLOBAL PROPERTY PM_SUBSYS_OUTPUT_PATH)
if (img_pm_subsys_path)
  foreach (pm_yml_path ${img_pm_subsys_path})
   list(GET img_pm_subsys_output_path 0 output_pm_yml_path)
   list(REMOVE_AT img_pm_subsys_output_path 0) 
  ...... and then use ${pm_yml_path} and ${output_pm_yml_path} as now.
```

this allow us to use the following when we require CMake >= 3.17
```
  foreach (pm_yml_path IN_ZIP_LISTS img_pm_subsys_path img_pm_subsys_output_path )
  ...  and then use ${pm_yml_path} and ${output_pm_yml_path} as now.
```

Reply:
Check it now :)

Reply:
Check it now